lib/fs/filesystem.js

```diff
@@ -125,8 +125,11 @@ class FileSystem {
     operationWrapper(this.rest, 'info', [path], (e, json) => {
       if (e) {
         if (e.statusCode === 404) {
-          // Negative cache.
-          this._addCache(path, e, 2);
+          // Negative cache, don't use the original (heavyweight) error instance.
+          const newError = new Error('File not found');
+          newError.statusCode = e.statusCode;
+          newError.message = e.message;
```

Comment:
Could this be an error that has a stack trace set as well that we would want to keep?

Reply:
I don't think so, I am only transforming 404 errors, during a `stat()` call. The reason for caching the `newError` is basically a flag that the item does not exist. I could use some other object like a sentinal etc, but storing the error allows me to return _it_ from cache during future `stat()` calls.

Reply:
By return, I mean pass to callback, I think most users of the api would just look for `if (e) ...` and maybe look at the `statusCode` etc.